packages/sanity/src/core/form/inputs/PortableText/Editor.tsx

```diff
@@ -51,9 +51,13 @@ interface EditorProps {
 }
 
 const renderDecorator: RenderDecoratorFunction = (props) => {
-  const {value, renderDefault, type, focused, selected} = props
-  const CustomComponent = type.components?.item
-  const rendered = renderDefault(props)
+  const {children, value, type, focused, selected} = props
+  const CustomComponent = type.component
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const RenderDefault = useCallback(
```

Comment:
why do we need to violate the rules of hooks here? Also unclear to me why  `renderDefault` is capitalized, is it intentional?

Reply:
No, this should not be like this 😅